docs/website/docs/dlt-ecosystem/verified-sources/pg_replication.md

```diff
@@ -0,0 +1,253 @@
+   ```sh
+   dlt pipeline <pipeline_name> show
+   ```
+   For example, the `pipeline_name` for the above pipeline example is `pg_replication_pipeline`, you may also use any custom name instead.
+
+
+   For more information, read the guide on [how to run a pipeline](https://dlthub.com/docs/walkthroughs/run-a-pipeline).
+    
+
+## Sources and resources
+
+`dlt` works on the principle of [sources](https://dlthub.com/docs/general-usage/source) and [resources](https://dlthub.com/docs/general-usage/resource).
+
+### Resource `replication_resource`
+
+This resource yields data items for changes in one or more Postgres tables.
+
+```py
+@dlt.resource(
+    name=lambda args: args["slot_name"] + "_" + args["pub_name"],
+    standalone=True,
+)
+def replication_resource(
+    slot_name: str,
+    pub_name: str,
+    credentials: ConnectionStringCredentials = dlt.secrets.value,
+    include_columns: Optional[Dict[str, Sequence[str]]] = None,
+    columns: Optional[Dict[str, TTableSchemaColumns]] = None,
+    target_batch_size: int = 1000,
+    flush_slot: bool = True,
+) -> Iterable[Union[TDataItem, DataItemWithMeta]]:
+    ...
+```
+
+`slot_name`: Replication slot name to consume messages.
+`pub_name`: Publication slot name to publish messages.
+
+`include_columns`: Maps table name(s) to sequence of names of columns to include in the generated data items. Any column not in the sequence is excluded. If not provided, all columns are included
+
+`columns`:  Maps table name(s) to column hints to apply on the replicated table(s)
+
+`target_batch_size`: Desired number of data items yielded in a batch. Can be used to limit the data items in memory.
+
+`flush_slot`:  Whether processed messages are discarded from the replication slot. The recommended value is "True".
+
+## Customization
+
+If you wish to create your own pipelines, you can leverage source and resource methods from this verified source.
+
+1. Define the source pipeline as:
+    
+   ```py
+   # Defining source pipeline
+   src_pl = dlt.pipeline(
+       pipeline_name="source_pipeline",
+       destination="postgres",
+       dataset_name="source_dataset",
+       dev_mode=True,
+   )
+   ```
+    
+   :::note 
+   You can configure and use the `get_postgres_pipeline()` function available in the `pg_replication_pipeline.py` file to achieve the same functionality. 
+   :::
+    
+2. Similarly, define the destination pipeline.
+    
+   ```py
+   dest_pl = dlt.pipeline(
+       pipeline_name="pg_replication_pipeline",
+       destination='duckdb',
+       dataset_name="replicate_single_table",
+       dev_mode=True,
+   )
+   ```
+    
+3. Define the slot and publication names as:
+    
+   ```py
+   slot_name = "example_slot"
+   pub_name = "example_pub"
+   ```
+    
+4. For the initial replication of the table to the destination, use the `init_replication` function:
+    
+   ```py
+   # requires the Postgres user to have the REPLICATION attribute assigned
+   initial_load = init_replication(  
```

Comment:
This is confusing. This is not doing an initial load. Initial loading requires setting `persist_snapshots` to `True`. See `replicate_with_initial_load()` in `pg_replication_pipeline.py`.

`init_replication` is short for "initialize replication", not "initial replication". A user can use this function to let `dlt` configure Postgres and make it ready for replication. Alternatively, a user can also do this outside of `dlt`. Using `init_replication` does not automatically mean you're doing an initial load.

Reply:
added some more contest